ui/app/store/actions.js

```diff
@@ -645,6 +645,89 @@ export function signPersonalMsg (msgData) {
   }
 }
 
+export function decryptMsgInline (msgData) {
+  log.debug('action - decryptMsgInline')
+  return (dispatch) => {
+    window.onbeforeunload = null
+    return new Promise((resolve, reject) => {
+      log.debug(`actions calling background.decryptMessageInline`)
+      background.decryptMessageInline(msgData, (err, newState) => {
+        log.debug('decryptMsg called back')
+        dispatch(updateMetamaskState(newState))
+        msgData = newState.unapprovedDecryptMsgs[msgData.metamaskId]
```

Comment:
I got an error here: `Error in event handler: TypeError: Cannot read property 'unapprovedDecryptMsgs' of undefined`

Reply:
Please provide more information about how I can reproduce this bug. Also, please watch the video about user flow in the test dapp.